winch/codegen/src/codegen/call.rs

```diff
@@ -165,7 +165,28 @@ impl<'a> FnCall<'a> {
 
     fn post_call<M: MacroAssembler>(&self, masm: &mut M, context: &mut CodeGenContext, size: u32) {
         masm.free_stack(self.call_stack_space + size);
-        context.drop_last(self.abi_sig.params.len());
+        // Only account for registers given that any memory entries
+        // consumed by the call (assigned to a register or to a stack
+        // slot) were freed by the previous call to
+        // `masm.free_stack`, so we only care about dropping them
+        // here.
+        //
+        // NOTE / TODO there's probably a path to getting rid of
```

Comment:
I'm planning on doing this refactor, but I didn't want to introduce it as part of this change.

Reply:
It's not entirely clear what's going on in the diff here (i.e. why things changed): is it that the registers weren't actually freed before?

Reply:
The registers were actually freed, implicitly as part of the previous implementation of the `drop_last` function, with the new implementation, the processing of certain values like registers and memory are explicit and up to the caller. I noticed that this change to the `drop_last` function has the potential to reduce a bit the boilerplate in calls, hence the `NOTE / TODO` comment. Hope this clarifies things up.